inst/examples/CommandBar.R

```diff
@@ -1,69 +1,91 @@
-        list(
-          key = "emailMessage",
-          text = "Email message",
-          iconProps = list(iconName = "Mail")
-        ),
-        list(
-          key = "calendarEvent",
-          text = "Calendar event",
-          iconProps = list(iconName = "Calendar")
+    CommandBarItem(
+      key = ns("newItem"),
+      text = "New",
+      cacheKey = "myCacheKey",
+      split = TRUE,
+      iconProps = list(iconName = "Add"),
+      subMenuProps = list(
+        items = list(
+          CommandBarItem(
+            key = ns("emailMessage"),
+            text = "Email message",
+            iconProps = list(iconName = "Mail")
+          ),
+          CommandBarItem(
+            key = ns("calendarEvent"),
+            text = "Calendar event",
+            iconProps = list(iconName = "Calendar")
+          )
         )
       )
+    ),
+    CommandBarItem(
+      key = ns("upload"),
+      text = "Upload",
+      iconProps = list(iconName = "Upload")
+    ),
+    CommandBarItem(
+      key = ns("share"),
+      text = "Share",
+      iconProps = list(iconName = "Share")
+    ),
+    CommandBarItem(
+      key = ns("download"),
+      text = "Download",
+      iconProps = list(iconName = "Download")
     )
-  ),
-  list(
-    key = "upload",
-    text = "Upload",
-    iconProps = list(iconName = "Upload")
-  ),
-  list(
-    key = "share",
-    text = "Share",
-    iconProps = list(iconName = "Share")
-  ),
-  list(
-    key = "download",
-    text = "Download",
-    iconProps = list(iconName = "Download")
   )
-)
+}
 
-farItems <- list(
-  list(
-    key = "tile",
-    text = "Grid view",
-    ariaLabel = "Grid view",
-    iconOnly = TRUE,
-    iconProps = list(iconName = "Tiles")
-  ),
+farItems <- function(ns) {
   list(
-    key = "info",
-    text = "Info",
-    ariaLabel = "Info",
-    iconOnly = TRUE,
-    iconProps = list(iconName = "Info")
+    CommandBarItem(
+      key = ns("tile"),
+      text = "Grid view",
+      ariaLabel = "Grid view",
+      iconOnly = TRUE,
+      iconProps = list(iconName = "Tiles")
+    ),
+    CommandBarItem(
+      key = ns("info"),
+      text = "Info",
+      ariaLabel = "Info",
+      iconOnly = TRUE,
+      iconProps = list(iconName = "Info")
+    )
   )
-)
-
+}
 
 ui <- function(id) {
   ns <- NS(id)
-  CommandBar(items = items, farItems = farItems)
+  tagList(
+    CommandBar(
+      items = items(ns),
+      farItems = farItems(ns)
+    ),
+    textOutput(ns("commandBarItems")),
+    CommandBar.shinyInput(
+      inputId = ns("commandBar"),
+      items = items(identity),
+      farItems = farItems(identity)
+    ),
+    textOutput(ns("commandBar"))
+  )
 }
 
 server <- function(id) {
-  moduleServer(id, function(input, output, session) { })
+  moduleServer(id, function(input, output, session) {
+    commandBarItemClicked <- reactiveVal()
+    observeEvent(input$newItem, commandBarItemClicked("newItem clicked"))
+    observeEvent(input$upload, commandBarItemClicked("upload clicked"))
```

Comment:
How about adding context to the message for the non-shinyInput observed items?

Mainly because it's not "dummy-proof" and it took me a bit to understand what's happening in the example.

```suggestion
    observeEvent(input$newItem, commandBarItemClicked("newItem clicked (needs to be explicitly observed)"))
    observeEvent(input$upload, commandBarItemClicked("upload clicked (needs to be explicitly observed)"))
```

Reply:
Thanks @averissimo! Good points as always 💯 

I added an issue #207 and improved the example